src/schema/posts.ts

```diff
@@ -2479,6 +2506,35 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return request.transaction;
     },
+    cancelPostBoost: async (
+      _,
+      { postId }: { postId: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      const post = await validatePostBoostPermissions(ctx, postId);
+
+      if (!post.flags?.boosted) {
+        throw new ValidationError('Post is not currently boosted');
+      }
+
+      const { userId } = ctx;
+
+      await ctx.con.transaction(async (entityManager) => {
+        await skadiBoostClient.cancelPostCampaign({
+          postId,
+          userId,
+        });
+
+        await entityManager
+          .getRepository(Post)
+          .update(
+            { id: postId },
+            { flags: updateFlagsStatement<Post>({ boosted: false }) },
+          );
```

Comment:
Better put skadi call at the end like we do with njord, it so that we first set boosted: false and then if skadi fails transaction will auto revert.

Like this skadi could go through but then DB fails, missed that in start post boost review as well.

Reply:
Good point! Let's do that.